include/libdnf-cli/argument_parser.hpp

```diff
@@ -33,56 +33,48 @@ namespace libdnf::cli {
 class ArgumentParser {
 public:
     /// Parent for all ArgumentsParser runtime errors.
-    class Exception : public RuntimeError {
-        using RuntimeError::RuntimeError;
+    class Error : public libdnf::Error {
+        using libdnf::Error::Error;
         const char * get_domain_name() const noexcept override { return "libdnf::cli::ArgumentParser"; }
-        const char * get_name() const noexcept override { return "Exception"; }
-        const char * get_description() const noexcept override { return "ArgumentParser exception"; }
+        const char * get_name() const noexcept override { return "Error"; }
     };
 
     /// Exception is generated when conflicting arguments are used together.
-    class Conflict : public Exception {
+    class Conflict : public Error {
```

Comment:
I see this has been a lot of work so I feel kinda bad, but I did mention this a week or two ago. I'm inclined to think it would be better to have the exceptions outside of the classes they are used in as an API consistency rule (because there will be quite a few exceptions that need to be outside of classes).

And I also think all exceptions should end with `Error`, e.g. here `ConflictError`, and this particular one doesn't seem descriptive enough, I think `ConflictingArgumentsError` would be better.